From 921676eac1476f30030e0a411871468a141e877b Mon Sep 17 00:00:00 2001
From: Serhii Kupriienko
Date: Tue, 27 Jan 2026 13:53:29 +0200
Subject: [PATCH] Skip pytest-jupyter-server plugin on py314

---
 tests/conftest.py | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/tests/conftest.py b/tests/conftest.py
index 440bcf9c0..603a9bfdf 100644
--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -15,7 +15,13 @@ from tests.extension.mockextensions.app import MockExtensionApp
 # dones not work in CI.
 os.environ["PYWINPTY_BACKEND"] = "1"
 
-pytest_plugins = ["jupyter_server.pytest_plugin"]
+
+import sys
+if sys.version_info < (3, 14):
+    pytest_plugins = ["jupyter_server.pytest_plugin"]
+else:
+    # pytest-jupyter not available for py314 yet
+    pytest_plugins = []
+
+
 def pytest_addoption(parser):
