From cee3b1a51febb42df63f82fc20091aaf21136c4f Mon Sep 17 00:00:00 2001
From: Serhii Kupriienko
Date: Tue, 27 Jan 2026 15:50:30 +0200
Subject: [PATCH] Fix tests

---
 tests/conftest.py | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/tests/conftest.py b/tests/conftest.py
index 440bcf9c0..678564d5a 100644
--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -15,7 +15,12 @@ from tests.extension.mockextensions.app import MockExtensionApp
 # dones not work in CI.
 os.environ["PYWINPTY_BACKEND"] = "1"
 
-pytest_plugins = ["jupyter_server.pytest_plugin"]
+import sys
+if sys.version_info < (3, 14):
+    pytest_plugins = ["jupyter_server.pytest_plugin"]
+else:
+    # pytest-jupyter not available for py314 yet
+    pytest_plugins = []
 
 
 def pytest_addoption(parser):
-- 
2.50.1 (Apple Git-155)

